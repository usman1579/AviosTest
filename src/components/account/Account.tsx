import React from 'react';
import { View, StyleSheet } from 'react-native';
import { useAppSelector } from '@/store/hooks';
import { selectMember } from '@/slices/auth';
import { AccontInfo } from './component';

export function Account() {
  const member = useAppSelector(selectMember);

  return (
    <View style={styles.container}>
      <AccontInfo headline={'FirstName'} value={member?.firstName} />
      <AccontInfo headline={'LastName'} value={member?.lastName} />
      <AccontInfo headline={'Email'} value={member?.email} />
      <AccontInfo headline={'Avious Points'} value={member?.avios} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'white',
    padding: 20,
  },
});
