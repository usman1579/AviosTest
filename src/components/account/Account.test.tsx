import React from 'react';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore, { MockStoreEnhanced } from 'redux-mock-store';
import { Account } from './Account';

const mockStore = configureStore<any>([]);

describe('Account component', () => {
  let store: MockStoreEnhanced<any, {}>;
  let component1: ReactTestRenderer;

  beforeEach(() => {
    store = mockStore({
      auth: {
        member: {
          firstName: 'John',
          lastName: 'Doe',
          email: 'john.doe@example.com',
          avios: 100,
        },
      },
    });

    component1 = renderer.create(
      <Provider store={store}>
        <Account />
      </Provider>,
    );
  });

  it('renders correctly', () => {
    const tree = component1.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Add more test cases as needed

  it('displays correct member information', () => {
    const instance = component1.root;

    // Replace with the actual selectors if needed
    const firstNameNode = instance.findByProps({ headline: 'FirstName' });
    const lastNameNode = instance.findByProps({ headline: 'LastName' });
    const emailNode = instance.findByProps({ headline: 'Email' });
    const aviosNode = instance.findByProps({ headline: 'Avious Points' });

    expect(firstNameNode.props.value).toEqual('John');
    expect(lastNameNode.props.value).toEqual('Doe');
    expect(emailNode.props.value).toEqual('john.doe@example.com');
    expect(aviosNode.props.value).toEqual(100);
  });
});
