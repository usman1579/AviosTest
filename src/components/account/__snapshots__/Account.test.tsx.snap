// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <View
    style={
      {
        "margin": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      FirstName
    </Text>
    <Text
      style={
        {
          "color": "grey",
          "fontSize": 16,
          "fontWeight": "500",
          "margin": 5,
          "marginLeft": 20,
        }
      }
    >
      John
    </Text>
  </View>
  <View
    style={
      {
        "margin": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      LastName
    </Text>
    <Text
      style={
        {
          "color": "grey",
          "fontSize": 16,
          "fontWeight": "500",
          "margin": 5,
          "marginLeft": 20,
        }
      }
    >
      Doe
    </Text>
  </View>
  <View
    style={
      {
        "margin": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Email
    </Text>
    <Text
      style={
        {
          "color": "grey",
          "fontSize": 16,
          "fontWeight": "500",
          "margin": 5,
          "marginLeft": 20,
        }
      }
    >
      john.doe@example.com
    </Text>
  </View>
  <View
    style={
      {
        "margin": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Avious Points
    </Text>
    <Text
      style={
        {
          "color": "grey",
          "fontSize": 16,
          "fontWeight": "500",
          "margin": 5,
          "marginLeft": 20,
        }
      }
    >
      100
    </Text>
  </View>
</View>
`;
